pylzy/lzy/api/v1/snapshot.py

```diff
@@ -21,23 +23,44 @@ class DataScheme:
     scheme_type: str
 
 
-@dataclasses.dataclass(frozen=True)
+@dataclasses.dataclass
 class SnapshotEntry:
     id: str
     name: str
     typ: Type
-    storage_uri: str
-    storage_name: str
     data_scheme: Schema
+    storage_name: str
+    _storage_uri: Union[str, None] = None
+    _data_hash: Union[str, None] = None
+
+    @property
+    def storage_uri(self) -> str:
+        if self._storage_uri is None:
+            raise ValueError(f"Storage uri for snapshot entry {id} is not set")
+        return cast(str, self._storage_uri)
+
+    @storage_uri.setter
+    def storage_uri(self, uri: str) -> None:
+        self._storage_uri = uri
+
+    @property
+    def data_hash(self) -> str:
+        if self._data_hash is None:
+            raise ValueError(f"Data hash for snapshot entry {id} is not set")
+        return cast(str, self._data_hash)
+
+    @data_hash.setter
+    def data_hash(self, hsh: str) -> None:
+        self._data_hash = hsh
 
 
 class Snapshot(ABC):  # pragma: no cover
     @abstractmethod
-    def create_entry(self, name: str, typ: Type, storage_uri: str) -> SnapshotEntry:
+    def create_entry(self, name: str, typ: Type) -> SnapshotEntry:
         pass
 
     @abstractmethod
-    def update_entry(self, entry_id: str, storage_uri: str) -> None:
+    def set_storage_uri_for_entry(self, entry_id: str, storage_uri_suffix: str) -> None:
```

Comment:
I don't like suffix here, because in this case, there is no single entity that is responsible for URI generation. Moreover, method name `set_storage_uri_for_entry` is misleading because only part of the URI is set here.

Maybe it is better to remove prefix field from Snapshot and set the whole URI on entry?

Reply:
in general I agree, uri construction should be better placed outside Snapshot